src/commonMain/kotlin/komem/litmus/runners/LitmusTestRunnerUtil.kt

```diff
@@ -0,0 +1,74 @@
+package komem.litmus.runners
+
+import komem.litmus.*
+import kotlin.time.Duration
+import kotlin.time.TimeSource
+
+fun LitmusTestRunner.runTest(
+    timeLimit: Duration,
+    params: RunningParams,
+    testProducer: () -> LitmusTest,
+): LitmusResult {
+    val results = mutableListOf<LitmusOutcomeInfo>()
+    val start = TimeSource.Monotonic.markNow()
+    while (start.elapsedNow() < timeLimit) {
+        val outcomes = runTest(params, testProducer)
+        val outcomeSetup = testProducer().getOutcomeSetup()
+        results.addAll(outcomes.groupToInfo(outcomeSetup))
+    }
+    return results.mergeOutcomes()
+}
+
+/*
+ Note: interprets AffinityMap as a sequence of smaller maps
```

Comment:
Block comments should have `*` at the beginning of each line.
https://google.github.io/styleguide/javaguide.html#s4.8.6-comments